### ISSUE TITLE
Textures missing on iOS in HTML5 export

### ISSUE THREAD

**Issue by combatwombat** [godotengine/godot#34766]
**Godot version:** 3.1.2.stable on macOS 14.6

**OS/device including version:** iOS 13.3 (iPhone XR, iPad 2018). Also reported on Android, but I have no details there.

**Issue description:** A MeshInstance with a SpatialMaterial and an Albedo ImageTexture renders fine in desktop browsers, but with a missing texture on iOS Safari.

**Steps to reproduce:** Create a MeshInstance with a texture, add a light and a camera. Export as HTML5. Test on iOS. Or look at the demo: https://combatwombat.github.io/godot-texture-test/

**Minimal reproduction project:** https://github.com/combatwombat/godot-texture-test

**Comment by Ajver** [comment]
I am running with the same problem and seems to I found something. Try to import texture in different preset mode 

(there are 4 presets available) 
![image](https://user-images.githubusercontent.com/40212297/71732062-a2ac2300-2e46-11ea-93c1-6b0e838ae7b1.png)

For me it works well with: 
* 2D, Detect 3D
* 2D

Probably 2D Pixel may works as well, but for me it sometimes does not work. 

Read about importing textures: https://docs.godotengine.org/en/3.1/getting_started/workflow/assets/importing_images.html

**Comment by alvaromunoz** [comment]
I don’t see a problem using iPhone XS Max iOS 13.3
![C31FDCA8-C9AA-44F0-AE3A-DFF0163087F0](https://user-images.githubusercontent.com/2722773/71745190-30d4d900-2e48-11ea-89bf-2899409d2584.png)

**Comment by combatwombat** [comment]
Ha, reimporting as "2D" worked, thanks @Ajver :)

It's still a bug I think but at least there's a workaround.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by kenan1099** [comment]
> Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

Yes, (above is iOS 14.2 on iPhone 7, below is firefox) 

![image](https://user-images.githubusercontent.com/37013292/102418937-2e976a80-4018-11eb-9587-7b7a5d12f875.png)
![Screenshot from 2020-12-17 03-27-11](https://user-images.githubusercontent.com/37013292/102418953-36570f00-4018-11eb-8863-2eab20b3e1d2.png)

**Comment by Calinou** [comment]
@kquote03 @combatwombat Can you reproduce this if you keep the default texture import settings but change Compress from **VRAM Compressed** to **Lossless**? We need to pinpoint the exact setting which causes this bug.

**Comment by kquote03** [comment]
![image](https://user-images.githubusercontent.com/37013292/170691536-90983bdf-bafa-40cd-8ee0-161c927220aa.png)
I changed this setting as instructed , then deleted the model and material file in my project, dragged and dropped the glb file into the godot project. Sadly this made no effect
![image](https://user-images.githubusercontent.com/37013292/170691941-1c60aa06-ae72-4fa8-be89-afe83e481a05.png)


> @kquote03 @combatwombat Can you reproduce this if you keep the default texture import settings but change Compress from **VRAM Compressed** to **Lossless**? We need to pinpoint the exact setting which causes this bug.

**Comment by kquote03** [comment]
For reference here's what it should look like:
![image](https://user-images.githubusercontent.com/37013292/170692390-e4feed42-80f3-4d06-86c7-181767dc9cb3.png)

Btw I tried the godot texture test website linked in the original comment, and for some reason it worked for me
![image](https://user-images.githubusercontent.com/37013292/170692506-0eb1e095-4622-4e8b-a0d7-3b522af51e5e.png)

Maybe it's a separate issue?

**Comment by Calinou** [comment]
> I changed this setting as instructed , then deleted the model and material file in my project, dragged and dropped the glb file into the godot project. Sadly this made no effect

You changed the Texture3D import default, which is used for 3D textures. Instead, you should have changed the import default on Texture2D or ImageTexture. A Texture3D is **not** a 2D texture applied on a 3D surface. It's a texture with actual 3-dimensional data (which you can view as a set of 2D texture layers).

**Comment by kquote03** [comment]
@Calinou Yeah I had a feeling I did something wrong, anyways I went and checked the texture import settings and it was already on lossless. Image and Bitmap import settings dont have a compression option

**Comment by Calinou** [comment]
> @Calinou Yeah I had a feeling I did something wrong, anyways I went and checked the texture import settings and it was already on lossless. Image and Bitmap import settings dont have a compression option

The Texture import defaults don't reflect flags that are automatically chosen when the texture is *detected* to be used in a 3D context. To force lossless compression on a texture used in a 3D context, you need to select it in the FileSystem dock and change its options in the Import dock.

**Comment by clayjohn** [comment]
I think the issue with older iphones is from using the sRGB option. sRGB should be diasbled and detect 3D should be turned off. This is why the 2D import default tends to work on these devices as well.

**Comment by kquote03** [comment]
@Calinou Alright I re-exported the model from blender in the collada format and re-imported using the following settings:
![image](https://user-images.githubusercontent.com/37013292/170737736-ca42606e-35ec-4cce-bba9-c863ea5ccd47.png)

Sadly it made no change on my iPhone

**Comment by Calinou** [comment]
> I think the issue with older iphones is from using the sRGB option. sRGB should be diasbled and detect 3D should be turned off. This is why the 2D import default tends to work on these devices as well.

I'm surprised sRGB textures is causing things to break, as I thought the conversion was purely a CPU-side operation performed at import-time.

Also, isn't the sRGB import option supposed to do nothing when using the GLES2 rendering backend?

**Comment by kquote03** [comment]
> I think the issue with older iphones is from using the sRGB option. sRGB should be diasbled and detect 3D should be turned off. This is why the 2D import default tends to work on these devices as well.

![image](https://user-images.githubusercontent.com/37013292/170738392-50579cab-4ccb-4013-9755-48cbca1aa321.png)

It's on newer iPhones as well, this was taken on an iPhone 11

**Comment by Calinou** [comment]
> It's on newer iPhones as well, this was taken on an iPhone 11

Which iOS version are you running on that iPhone 11?

**Comment by kquote03** [comment]
@Calinou 15.5
